client/app/components/FavoritesControl.jsx

```diff
@@ -0,0 +1,89 @@
+import { isFunction } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { $rootScope } from '@/services/ng';
+
+function toggleItem(event, item, callback) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (item) {
+    if (item.is_favorite) {
+      item.$unfavorite().then(() => {
+        item.is_favorite = false;
+        $rootScope.$broadcast('reloadFavorites');
+        if (isFunction(callback)) {
+          callback();
+        }
+      });
+    } else {
+      item.$favorite().then(() => {
+        item.is_favorite = true;
+        $rootScope.$broadcast('reloadFavorites');
+        if (isFunction(callback)) {
+          callback();
+        }
+      });
+    }
+  }
+}
+
+export function FavoritesControl({ item, onChange }) {
+  const icon = item.is_favorite ? 'fa-star' : 'fa-star-o';
+  const title = item.is_favorite ? 'Remove from favorites' : 'Add to favorites';
```

Comment:
A trick I like to do to avoid duplicate condition:
```js
const [icon, title] = item.is_favorite
  ? ['Remove from favorites', 'fa-star']
  : ['Add to favorites', 'fa-star-o'];
```

Not really suggesting to swap for this. Just another way of doing the same thing 😬

Reply:
IMHO both variants are more or less equal, so let's keep my one :wink: